### ISSUE TITLE
LIKE() not working as expected if expression contains asterisk character

### ISSUE THREAD

**Issue by Simran-B** [arangodb/arangodb#4919]
_Discovered by [Rupak Karna](https://stackoverflow.com/questions/49401667/arangodb-like-function-doesnt-seem-to-filter-out-values-containing)_

## My environment

* __ArangoDB version__: 3.3.4
* __Storage engine__: RocksDB
* __Deployment mode__: single server
* __Operating system__: Windows 10

## Steps to reproduce

Run this with arangosh:

```js
var collname = "likebug";
db._drop(collname)
var coll = db._create(collname);
coll.save([ { "pattern": "abcde*fgh" }, { "pattern": "test*new" }, { "pattern": "abc%^def" } ]);

["*", "%*%", "%\\*%", "test*n%", "test*new", "test\\*new"].forEach(expr => {
    print()
    print("LIKE: " + JSON.stringify(expr));
    print(db._query(`
        LET native = MERGE(FOR doc IN @@collname RETURN { [doc.pattern]: LIKE(doc.pattern, @expr)})
        LET v8 = MERGE(FOR doc IN @@collname RETURN { [doc.pattern]: NOOPT(V8(LIKE(doc.pattern, @expr)))})
        RETURN {native, v8}`, {"@collname": collname, expr}).toArray());
});
```

__Problem__:

The results of the native implementation of LIKE() does not match the behavior of the V8 version. The `*` character seems to have some special behavior, but it can not be worked around with backslash escaping. `%*%` does not even produce a boolean, but `null`.

__Expected result__:

C++ LIKE() should return the same results as V8, with no need to escape `*` and always returning either `true` or `false`.

```
LIKE: "*"
[
  {
    "native" : {
      "abcde*fgh" : false,
      "test*new" : false,
      "abc%^def" : false
    },
    "v8" : {
      "abcde*fgh" : false,
      "test*new" : false,
      "abc%^def" : false
    }
  }
]

LIKE: "%*%"
[
  {
    "native" : {
      "abcde*fgh" : null,
      "test*new" : null,
      "abc%^def" : null
    },
    "v8" : {
      "abcde*fgh" : true,
      "test*new" : true,
      "abc%^def" : false
    }
  }
]

LIKE: "%\\*%"
[
  {
    "native" : {
      "abcde*fgh" : true,
      "test*new" : true,
      "abc%^def" : true
    },
    "v8" : {
      "abcde*fgh" : true,
      "test*new" : true,
      "abc%^def" : false
    }
  }
]

LIKE: "test*n%"
[
  {
    "native" : {
      "abcde*fgh" : false,
      "test*new" : false,
      "abc%^def" : false
    },
    "v8" : {
      "abcde*fgh" : false,
      "test*new" : true,
      "abc%^def" : false
    }
  }
]

LIKE: "test*new"
[
  {
    "native" : {
      "abcde*fgh" : false,
      "test*new" : false,
      "abc%^def" : false
    },
    "v8" : {
      "abcde*fgh" : false,
      "test*new" : true,
      "abc%^def" : false
    }
  }
]

LIKE: "test\\*new"
[
  {
    "native" : {
      "abcde*fgh" : false,
      "test*new" : false,
      "abc%^def" : false
    },
    "v8" : {
      "abcde*fgh" : false,
      "test*new" : true,
      "abc%^def" : false
    }
  }
]
```

**Comment by ObiWahn** [comment]
https://github.com/arangodb/arangodb/pull/5239

**Comment by maxkernbach** [comment]
Fixed since version 3.3.9 with https://github.com/arangodb/arangodb/pull/5239
